src/templatizer/array-selector.html

```diff
@@ -0,0 +1,315 @@
+        <div>First name: <span>{{item.first}}</span></div>
+        <div>Last name: <span>{{item.last}}</span></div>
+        <button on-click="toggleSelection">Select</button>
+    </template>
+
+    <array-selector id="selector" items="{{employees}}" selected="{{selected}}" multi toggle></array-selector>
+
+    <div> Selected employees: </div>
+    <template is="dom-repeat" items="{{selected}}">
+        <div>First name: <span>{{item.first}}</span></div>
+        <div>Last name: <span>{{item.last}}</span></div>
+    </template>
+
+  </template>
+
+  <script>
+    Polymer({
+      is: 'employee-list',
+      ready() {
+        this.employees = [
+            {first: 'Bob', last: 'Smith'},
+            {first: 'Sally', last: 'Johnson'},
+            ...
+        ];
+      },
+      toggleSelection(e) {
+        var item = this.$.employeeList.itemForElement(e.target);
+        this.$.selector.select(item);
+      }
+    });
+  </script>
+
+</dom-module>
+```
+-->
+
+<link rel="import" href="../../polymer-element.html">
+<link rel="import" href="../utils/utils.html">
+<link rel="import" href="../utils/array-splice.html">
+
+<script>
+(function() {
+
+  let ArraySelectorMixin = Polymer.Utils.dedupingMixin(superClass => {
+
+    return class extends superClass {
+
+      static get config() {
+      
+        return {
+
+          properties: {
+
+            /**
+             * An array containing items from which selection will be made.
+             */
+            items: {
+              type: Array,
+            },
+
+            /**
+             * When `true`, multiple items may be selected at once (in this case,
+             * `selected` is an array of currently selected items).  When `false`,
+             * only one item may be selected at a time.
+             */
+            multi: {
+              type: Boolean,
+              value: false,
+            },
+
+            /**
+             * When `multi` is true, this is an array that contains any selected.
+             * When `multi` is false, this is the currently selected item, or `null`
+             * if no item is selected.
+             */
+            selected: {
+              type: Object,
+              notify: true
+            },
+
+            /**
+             * When `multi` is false, this is the currently selected item, or `null`
+             * if no item is selected.
+             */
+            selectedItem: {
+              type: Object,
+              notify: true
+            },
+
+            /**
+             * When `true`, calling `select` on an item that is already selected
+             * will deselect the item.
+             */
+            toggle: {
+              type: Boolean,
+              value: false
+            }
+
+          },
+
+          observers: ['_updateSelection(multi, items.*)'],
+
+        }
+
+      }
+
+      _updateSelection(multi, itemsInfo) {
+        if (itemsInfo.path == 'items') {
+          let newItems = itemsInfo.base;
```

Comment:
This observer might run even when `newsItems` is `undefined` causing a:
`Uncaught TypeError: Cannot read property 'length' of undefined`
from `calculateSplices (array-splice.html:251)`